src/SSH.ts

```diff
@@ -0,0 +1,112 @@
+import { ArgumentError } from "./errors/ArgumentError";
+import { RequestClient } from "./internal/RequestClient";
+
+type GenerateSSHCredentialsResponse = {
+  lease_id: string;
+  lease_duration: number;
+  renewable: boolean;
+  data: {
+    ip: string;
+    key: string;
+    key_type: "otp" | "ca";
+    port: number;
+    username: string;
+  }
+}
+
+type ListRolesByIpResponse = {
+  lease_id: string;
+  lease_duration: number;
+  renewable: boolean;
+  data: {
+    roles: string[];
+  }
+}
+
+type VerifySSHOTPResponse = {
+  lease_id: string;
+  lease_duration: number;
+  renewable: boolean;
+  data: {
+    ip: string;
+    username: string;
+  }
+}
+
+export class SSH {
+  constructor(
+    private readonly mountPath: string,
+    private readonly client: RequestClient
+  ) {
+    if (mountPath === "") throw new ArgumentError("mountPath");
```

Comment:
should also check for null/undefined case because this can still be used in js file